jose/jwk.py

```diff
@@ -209,43 +212,65 @@ def __init__(self, key, algorithm):
                     raise JWKError(e)
                 return
 
-            try:
-                self.prepared_key = RSA.importKey(key)
-            except Exception as e:
-                raise JWKError(e)
+            for method, is_private in [(load_der_public_key, False), (load_pem_public_key, False),
+                                       (load_pem_private_key, True), (load_der_private_key, True)]:
+                try:
+                    if is_private:
+                        self.prepared_key = method(
+                            key, password=None, backend=default_backend())
+                    else:
+                        self.prepared_key = method(
+                            key, backend=default_backend())
+                    break
+                except Exception as e:
+                    continue
+            else:
+                raise JWKError(
+                    'Could not load key using the available methods.')
+
             return
 
         raise JWKError('Unable to parse an RSA_JWK from key: %s' % key)
 
     def _process_jwk(self, jwk_dict):
         if not jwk_dict.get('kty') == 'RSA':
-            raise JWKError("Incorrect key type.  Expected: 'RSA', Recieved: %s" % jwk_dict.get('kty'))
+            raise JWKError(
+                "Incorrect key type.  Expected: 'RSA', Recieved: %s" % jwk_dict.get('kty'))
 
         e = base64_to_long(jwk_dict.get('e', 256))
         n = base64_to_long(jwk_dict.get('n'))
 
-        self.prepared_key = RSA.construct((n, e))
+        self.prepared_key = rsa.RSAPublicNumbers(
+            e, n).public_key(default_backend())
         return self.prepared_key
 
     def _process_cert(self, key):
-        pemLines = key.replace(b' ', b'').split()
-        certDer = base64url_decode(b''.join(pemLines[1:-1]))
-        certSeq = DerSequence()
-        certSeq.decode(certDer)
-        tbsSeq = DerSequence()
-        tbsSeq.decode(certSeq[0])
-        self.prepared_key = RSA.importKey(tbsSeq[6])
+
+        certSeq = x509.load_pem_x509_certificate(key, default_backend())
+        self.prepared_key = certSeq.public_key()
+
         return
 
     def sign(self, msg):
         try:
-            return PKCS1_v1_5.new(self.prepared_key).sign(self.hash_alg.new(msg))
+            return self.prepared_key.sign(
+                msg,
+                padding.PKCS1v15(),
+                self.hash_alg(),
+            )
         except Exception as e:
             raise JWKError(e)
 
     def verify(self, msg, sig):
         try:
-            return PKCS1_v1_5.new(self.prepared_key).verify(self.hash_alg.new(msg), sig)
+            self.prepared_key.verify(
+                sig,
+                msg,
+                padding.PKCS1v15(),
+                self.hash_alg(),
```

Comment:
One thing to note is that I'm not completely sure what happens when the hash_alg is not one of the SHA* hashes. Maybe some feedback on that?